### DISCUSSION TITLE
Desktop solution to replace electron

### DISCUSSION THREAD

**Discussion by orange-pig** [eclipse-theia/theia#13156] [Q&A]
Does anyone have any experience trying to replace theia's electron with another implementation on the desktop?

**Comment by msujew** [comment]
@orange-pig I don't think there will be anyone with experiences using Theia using something other than Electron. As far as I know, the Theia backend doesn't run outside of node, and Electron is the only desktop-framework solution that comes with node by default. Everything else (like Tauri) requires a non-node based backend, i.e. Rust for Tauri.

**Comment by orange-pig** [comment]
I want to achieve web page rendering through cef, but theia also has a backend. I don't know if it is possible to achieve the same effect of electron by embedding a node.js in the framework and support extension. 

Electron is very powerful in toC applications, but it is too large, consumes resources and is inefficient, which also makes it more difficult to apply to toB software.

Therefore, wanting to use the powerful functions of theia to in a prefermonce software has given rise to some unpopular hybrid solutions.

I think like Qt+lifcef+node, or Tauri + node, and others. If no one has explored this path, it will be a difficult process.

**Reply by msujew** [reply]
Given that the Electron parts of Theia are pretty heavily embedded into Theia itself, [see this whole directory](https://github.com/eclipse-theia/theia/tree/master/packages/core/src/electron-main), I'm not sure this is easily possible. You might be able to trick Theia into thinking it's running as a browser app on desktop, but then you lose a lot of functionality that's only available for Electron.

**Reply by orange-pig** [reply]
Yes, it's not easily.  It is not simple and meaningless to simulate an electron, after i looking at some alternative libraries for electron. I don’t know if Theia plans to provide a front-end-only mode. The back-end can be implemented using any other technology. This may be a feasible hybrid solution on the desktop and the web.
